streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/bolt/rules/WindowRulesBolt.java

```diff
@@ -87,43 +84,61 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
     public void execute(TupleWindow inputWindow) {
         ++windowId;
         LOG.debug("Window activated, window id {}, number of tuples in window {}", windowId, inputWindow.get().size());
-        List<Tuple> curGroup = new ArrayList<>();
+
+        Map<String, Tuple> eventIdToTupleMap = new HashMap<>();
         try {
             StreamlineEvent event;
             for (Tuple input : inputWindow.get()) {
                 if ((event = getStreamlineEventFromTuple(input)) != null) {
                     LOG.debug("++++++++ Executing tuple [{}] which contains StreamlineEvent [{}]", input, event);
-                    processAndEmit(event, curGroup);
-                    curGroup.add(input);
+                    eventIdToTupleMap.put(event.getId(), input);
+                    processAndEmit(event, eventIdToTupleMap);
                 }
             }
             // force evaluation of the last group by
-            processAndEmit(GROUP_BY_TRIGGER_EVENT, curGroup);
+            processAndEmit(GROUP_BY_TRIGGER_EVENT, eventIdToTupleMap);
+
+            // current group is processed and result emitted
+            eventIdToTupleMap.clear();
         } catch (Exception e) {
             collector.reportError(e);
-            LOG.debug("", e);                        // useful to debug unit tests
+            LOG.error("", e);
```

Comment:
SInce we are going to provide the feature to search the log, better to leave log message whenever it is informative.